bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureMiceDetectionHandler.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.verisure.internal.handler;
+
+import static org.openhab.binding.verisure.internal.VerisureBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import javax.measure.quantity.Temperature;
+import javax.measure.quantity.Time;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection.Detection;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection.Mouse;
+import org.openhab.binding.verisure.internal.model.VerisureThing;
+
+/**
+ * Handler for the Mice Detection thing type that Verisure provides.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class VerisureMiceDetectionHandler extends VerisureThingHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_MICE_DETECTION);
+
+    public VerisureMiceDetectionHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public synchronized void update(@Nullable VerisureThing thing) {
+        logger.debug("update on thing: {}", thing);
+        if (thing != null && SUPPORTED_THING_TYPES.contains(thing.getThingTypeUID())) {
+            updateStatus(ThingStatus.ONLINE);
+            VerisureMiceDetection obj = (VerisureMiceDetection) thing;
+            updateMiceDetectionState(obj);
+        } else {
+            logger.warn("Can't handle this thing typeuid: {}", getThing().getThingTypeUID());
+        }
+    }
+
+    private void updateMiceDetectionState(VerisureMiceDetection miceDetectionJSON) {
+        Mouse mouse = miceDetectionJSON.getData().getInstallation().getMice().get(0);
```

Comment:
Why are you only using the Mouse at index 0? What about the rest of them?

Reply:
They got scared! :-)
The binding has already handled the rest of the mice when parsing the result of the API query, and internally put them as unique things in the Session list.
So in this method there is only one mouse in the list to handle.